packages/studio-plugin/src/orchestrators/ParsingOrchestrator.ts

```diff
@@ -194,7 +212,11 @@ export default class ParsingOrchestrator {
     });
 
     if (absPath.startsWith(this.paths.components)) {
-      const componentFile = new ComponentFile(absPath, this.project);
+      const componentFile = new ComponentFile(
+        absPath,
+        this.project,
+        this.getComponentDependencyTree(absPath)
```

Comment:
I think the way in which we create the Dependency Tree in ParsingOrchestrator is a bit convoluted. Here, we're trying to create the `ComponentFile` without the requisite information (the Component's Dependency tree). So we create it on the fly and update a global variable (`this.dependencyTree`). There are two distinct places where we invoke `updateDependencyTree`. 

As an alternative, could we first create the full `dependencyTree` and then run `initFilepathToFileMetadata`? Or would that cause a performance hit?

Reply:
We currently initialize the `dependencyTree` as a part of `initFilepathToFileMetadata`, and I can move `updateDependencyTree` before we `getFileMetadata` so it should have the requisite information by the time it gets to it when we first initialize!  However, we probably won't be able to avoid using some sort of "on the fly" tree updating, since we support HMR.  This is important in the case when a developer adds a new component file and HMR makes it appear instantly in Studio.  Since we won't have that file's dependency tree, we'll need to generate it on the spot.

Reply:
Update to this-- we update dependency tree in `getFileMetadata`, since that is called in the `filepathToFileMetadata` initializer, so this initializes the dependency tree in the beginning.  Another reason we want to call this here is because if the user were to update any files with new CSS, we would want a new, updated dependency tree to make sure that their CSS imports are up to date.   This also just happens simplifies the code!